QModManager/Patcher.cs

```diff
@@ -266,7 +260,75 @@ internal static bool LoadMod(QMod mod)
 
         #endregion
 
-        #region LoadOrder
+        #region Game detection
+
+        internal enum Game
+        {
+            Subnautica,
+            BelowZero,
+        }
+
+        internal static Game game;
+
+        internal static bool DetectGame()
+        {
+            bool sn = Directory.GetFiles(Environment.CurrentDirectory, "Subnautica.exe", SearchOption.TopDirectoryOnly).Length > 0
+                || Directory.GetFiles(Environment.CurrentDirectory, "Subnautica.app", SearchOption.TopDirectoryOnly).Length > 0
+                || Directory.GetDirectories(Environment.CurrentDirectory, "Subnautica.app", SearchOption.TopDirectoryOnly).Length > 0;
+            bool bz = Directory.GetFiles(Environment.CurrentDirectory, "SubnauticaZero.exe", SearchOption.TopDirectoryOnly).Length > 0
+                || Directory.GetFiles(Environment.CurrentDirectory, "SubnauticaZero.app", SearchOption.TopDirectoryOnly).Length > 0
+                || Directory.GetDirectories(Environment.CurrentDirectory, "SubnauticaZero.app", SearchOption.TopDirectoryOnly).Length > 0;
+
+            if (sn && !bz) game = Game.Subnautica;
+            else if (bz && !sn) game = Game.BelowZero;
+            else if (sn && bz)
+            {
+                Console.WriteLine("[QModManager] A fatal error has occurred.");
+                Console.WriteLine("Both Windows and Mac files detected!");
+                Console.WriteLine("Is this a Windows or a Mac environment?");
+                Console.WriteLine();
+                return false;
+            }
+            else
+            {
+                Console.WriteLine("[QModManager] A fatal error has occurred.");
+                Console.WriteLine("No Subnautica executable was found!");
+                Console.WriteLine();
+                return false;
+            }
+            return true;
+        }
+
+        internal static void DisableNonApplicableMods()
+        {
+            List<QMod> nonApplicableMods = new List<QMod>();
+            sortedMods = sortedMods.Where(mod =>
+            {
+                if (mod.Game == game) return true;
+
+                if (!nonApplicableMods.Contains(mod)) nonApplicableMods.Add(mod);
+                if (!erroredMods.Contains(mod)) erroredMods.Add(mod);
+                return false;
+
+            }).ToList();
+
+            Console.WriteLine($"\nQMOD ERROR: The following {GetOtherGame()} mods were not loaded because {game.ToString()} was detected:");
```

Comment:
Log as a Warning, not an Error.

Reply:
If there are any errors with LoadOrder/Dependencies, it's logged as an error

Reply:
Semantics. The mod loader here correctly identifies that the mod shouldn't be loaded.  
It didn't encounter an error while trying to load it.  
And since it won't be loaded, it shouldn't cause issues for the player.

Hence why I would call this a warning. There are actions the user could take, but ultimately, things can proceed as normal.

Reply:
Addressed in commit fc9d40a